### ISSUE TITLE
RxCollection.findOne().exec() returns deleted document while find().exec() not

### ISSUE THREAD

**Issue by Albert-Gao** [pubkey/rxdb#2071]
```savascript
// create a record 
await rxdb.companies.insert({
              id: '22 ',
              name: 'super man',
              ceo_id: '11-22-33',
              created_at: '2020-03-20T09:37:53.696303Z',
})

// delete it
const doc = await rxdb.companies.findOne()
await doc.remove()

// now find() returns empty array
const docs = await rxdb.companies.find().exec()
alert(JSON.stringify(docs))

// findOne() still returns the deleted object
const doc = await rxdb.companies.findOne().exec()
alert(JSON.stringify(doc))
alert(doc.deleted) // true
```

so, findOne and find seems to return inconsistent result
Do I miss something?

**Comment by pubkey** [comment]
@Albert-Gao I added a test but everything looks fine. Can you check and maybe modify the test to reproduce the wrong behavior?

**Comment by pubkey** [comment]
This might be fixed with version `9.0.0`. I refactored much stuff with the event propagation and added more tests to ensure everything works. Especially timing related things.

**Comment by pubkey** [comment]
I'm closing this, I think is is fixed with 9.0.0 please try it out.
